heka/sandbox/filters/telemetry_latency.lua

```diff
@@ -0,0 +1,224 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+1) What's the distribution of the latency between the subsession ping creation
+   time and the server reception time in the past 24h?
+
+2) What's the distribution of the latency between the subsession ping creation time
+   and the server reception time in the past 7 days?
+
+3) What’s the delay in hours for active profiles between the date a ping was created 
+   and the time it was received on our servers? When a new submission for an
+   active profile is received, the latency between the ping creation date and the 
+   reception time is computed. Periodically, a histogram of the latencies is plot.
+
+4) How many days do we need to look back to see at least one submission for k% of 
+   active profiles? Each active profile is associated to the date for which a submission 
+   was last received on our servers. Periodically, we compute for all profiles the 
+   difference between the current date and the date of reception and finally plot 
+   a histogram of the differences expressed in number of days.
+
+Note that:
+   - As timeseries of histograms or heatmaps are not supported by the Heka plotting
+     facilities, only the median and some other percentiles are being output.
+
+   - An active profile is one who has used the browser in the last six weeks (42 days)
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [TelemetryLatency]
+    type = "SandboxFilter"
+    filename = "lua_filters/telemetry_latency.lua"
+    message_matcher = "Type == 'telemetry' && Fields[docType] == 'main'"
```

Comment:
You should accept on version and sample_id in the matcher.  It will save a trip down to the sandbox and the reads+test can be removed from the plugin.